renku/core/management/clone.py

```diff
@@ -33,23 +34,46 @@ def clone(
     skip_smudge=True,
     recursive=True,
     depth=None,
-    progress=None
+    progress=None,
+    config=None,
+    raise_git_except=False,
 ):
     """Clone Renku project repo, install Git hooks and LFS."""
     from renku.core.management.client import LocalClient
 
     path = path or '.'
+
+    if isinstance(path, Path):
+        path = str(path)
+
     # Clone the project
     if skip_smudge:
         os.environ['GIT_LFS_SKIP_SMUDGE'] = '1'
+
     try:
         repo = Repo.clone_from(
             url, path, recursive=recursive, depth=depth, progress=progress
         )
     except GitCommandError as e:
-        raise errors.GitError(
-            'Cannot clone remote Renku project: {}'.format(url)
-        ) from e
+        if not raise_git_except:
```

Comment:
if `raise_git_except` is `False`, it raises a git exception? Shouldn't it be the other way around?

Reply:
Maybe I could improve the wording here. The flag denotes if we should raise the vanilla git exception or raise renku wrapped one. 

Let me try to explain, so one is vanilla git exception and the other is renku wrapped git exception. The wrapped one doesn't really fit the service cause the message attribute is not set within the `__caused__` so I cannot access the explanation part for the reason of the exception (to return it as a response to denote what went wrong). So we need both or find a way to propagate the reason. I can open a ticket for this and look into more detail.